cmd/notification.go

```diff
@@ -238,22 +238,22 @@ func (sys *NotificationSys) initListeners(objAPI ObjectLayer, bucketName string)
 		return err
 	}
 
-	return saveConfig(objAPI, configFile, data)
+	return saveConfig(ctx, objAPI, configFile, data)
```

Comment:
@balamurugana, readConfig and saveConfig do not expose the minio meta directories. This PR is just abstracting .minio config directory and location of listener and notification config from the s3 layer. If these functions were pulled into the fs and xl layer, it would lead to code duplication.

Reply:
it seems i wasn't clearly said the problem.

methods in `NotificationSys` should not use or know the knowledge of object's layer entities like `.minio.sys`, `listener.json`, `notification.xml`, `policy.json` etc.

Based on this I was saying we need to have new methods `objAPI.GetBucketListenerConfig()` and `objAPI.SetBucketListenerConfig()` and I requested to check with @abperiasamy about introducing new methods.

Reply:
ok, will check with @abperiasamy tomorrow and update PR accordingly.